### PR TITLE
ci: Add dependabot CVE checks

### PR DESCRIPTION
**Pull request by eiffel-fl**
Hi.

This PR does 3 things:

1. It enables `dependabot` to check our `golang` dependencies do not contain CVE daily as suggested by Burak.
1. It enables `dependabot` to check our `docker` dependencies do not contain CVE daily as suggested by Burak.
1. When a PR is opened, it adds a check to ensure the dependencies added by this PR do not container CVE.

Best regards.

### CODE REVIEW

**Review by burak-ok — APPROVED** [review]
LGTM.

But now `CodeQL` fails the CI for this PR xD

File: .github/workflows/inspektor-gadget.yml [hunk 1]
```diff
@@ -170,6 +172,9 @@ jobs:
       # ALERT This action must be run after code was checkout otherwise it will
       # not find this file.
       uses: ./.github/actions/install-debian-packages
+    - uses: github/codeql-action/init@v2
+      with:
+        languages: go, cpp
```
Comment (mauriciovasquezbernal) [inline .github/workflows/inspektor-gadget.yml:186]: Why cpp?

**Review by mauriciovasquezbernal — COMMENTED** [review]
This looks good. Do you think 1day is the correct value for the interval? Won't it spam us too much with dependency update PRs?

File: .github/workflows/inspektor-gadget.yml (hunk 1 above)
Comment (burak-ok) [reply .github/workflows/inspektor-gadget.yml:186]: `codeql` summarizes C and C++ into one category: C++ https://codeql.github.com/docs/codeql-overview/supported-languages-and-frameworks/

Ofc the question is still if it even can understand the our headers and ebpf source at all

Comment (mauriciovasquezbernal) [reply .github/workflows/inspektor-gadget.yml:186]: Yeah, I don't think it'll provide anything related to bpf. I agree that we can merge and check what happens.

**Pull request merged**